src/lib.rs

```diff
@@ -1790,6 +1824,14 @@ impl Build {
             cmd.push_cc_arg(warnings_to_errors_flag);
         }
 
+        for flag in self.c_flags.iter() {
+            cmd.c_args.push((**flag).into());
+        }
+
+        for flag in self.cpp_flags.iter() {
+            cmd.cpp_args.push((**flag).into());
+        }
+
```

Comment:
I think you need to do this in `get_base_compiler`, there are some functions which call it directly.